app/backend/src/tests/test_search.py

```diff
@@ -158,6 +159,54 @@ def test_user_filter_meetup_status(db):
         assert [result.user.user_id for result in res.results] == [user_does_not_want_to_meet.id]
 
 
+def test_user_filter_language(db):
+    """
+    Make sure the language filter returns the rigth profiles
+    """
+
+    user_with_danish_fluent, token11 = generate_user()
+    user_with_czech_conversational, token12 = generate_user()
+
+    with session_scope() as session:
+        existing_language = session.query(Language).filter_by(code="dan").one_or_none()
+
+        if not existing_language:
+            # Insert the language record if it doesn't exist
+            session.add(Language(code="dan", name="Danish"))
+
+        existing_language = session.query(Language).filter_by(code="ces").one_or_none()
+
+        if not existing_language:
+            session.add(Language(code="ces", name="Czech"))
```

Comment:
No need to do this: these are defined in `languages.json` and are included in the `db` test dependency

Reply:
when i dont add the condition, it throws me an foreign key violation error. something like this 

`E       sqlalchemy.exc.IntegrityError: (psycopg2.errors.ForeignKeyViolation) insert or update on table "language_abilities" violates foreign key constraint "fk_language_abilities_language_code_languages"
E       DETAIL:  Key (language_code)=(dan) is not present in table "languages".
E       
E       [SQL: INSERT INTO language_abilities (user_id, language_code, fluency) SELECT p0::BIGINT, p1::VARCHAR, p2::languagefluency FROM (VALUES (%(user_id__0)s, %(language_code__0)s, %(fluency__0)s, 0), (%(user_id__1)s, %(language_code__1)s, %(fluency__1)s, 1)) AS imp_sen(p0, p1, p2, sen_counter) ORDER BY sen_counter RETURNING language_abilities.id, language_abilities.id AS id__1]
E       [parameters: {'fluency__0': 'fluent', 'user_id__0': 1, 'language_code__0': 'dan', 'fluency__1': 'conversational', 'user_id__1': 2, 'language_code__1': 'ces'}]
E       (Background on this error at: https://sqlalche.me/e/20/gkpj)
`

Reply:
Ahh sorry, my bad. Try using some from this list:

https://github.com/Couchers-org/couchers/blob/0fb8910a3bc0c3cf46b8251fdda7d1e7b12d9e78/app/backend/src/tests/test_fixtures.py#L178

Reply:
dome, thanks.